File: Python_Programming/Problems_on_Numbers/SunnyNumber.py
'''
    Description : program  to check the number is Sunny number or not 
                EX :- number + 1 is  perfect square number than its sunny number
                
    Date :- 21/03/2026
    
    Author : Rekha Shankarlal Kumawat
    
'''
import math 

def SunnyNumber(iNo):

    Ans = 0

    iNo = iNo +1

    Ans = int(math.sqrt(iNo))
    return(iNo == (Ans*Ans))

def main():
    
    iValue = 0
    bRet = False 

    print("Enter the Number :- ")
    iValue = int(input())
    
    bRet =  SunnyNumber(iValue)

    if(bRet == True):
        print(iValue ," is a Sunny Number\n")
    else:
        print(iValue ," is not a Sunny Number\n")

if __name__ == "__main__":
    main()

File: C_Program/Problem_on_Numbers/SunnyNumber.c
//////////////////////////////////////////////////////////////////////////////////////////////
//
//required header file
//
//////////////////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdbool.h>
#include<math.h>

//////////////////////////////////////////////////////////////////////////////////////////////
//
//  Function Name : SunnyNumber()
//  Description :   Check wheather the number is Sunny Number or Not
//  Input :         Integer
//  output :        bool
//  Author :        Rekha Shankarlal Kumawat
//  Date :          22/03/2026
//
//////////////////////////////////////////////////////////////////////////////////////////////


bool SunnyNumber(int iNo)
{
    int Ans = 0;

    iNo++;

    Ans = (int)sqrt(iNo);
    
    return(iNo == (Ans*Ans));

}
int main()
{
    int iValue = 0;
    bool bRet = false ;

    printf("Enter the Number :- ");
    scanf("%d" , &iValue);
    
    bRet =  SunnyNumber(iValue);

    if(bRet == true)
    {
        printf("%d is a Sunny Number\n", iValue);
    }
    else
    {
        printf("%d is not a Sunny Number\n", iValue);
    }

    return 0 ;
}